uni/lib/controller/background_workers/notifications.dart

```diff
@@ -0,0 +1,150 @@
+    if(test){
+      displayNotification(await buildNotificationContent(session), localNotificationsPlugin);
+    }
+  }
+}
+
+class NotificationManager{
+
+
+  static final FlutterLocalNotificationsPlugin _localNotificationsPlugin = FlutterLocalNotificationsPlugin();
+
+  static bool _initialized = false;
+
+  static const Duration _notificationWorkerPeriod = Duration(hours: 1);
+
+
+  static Future<void> updateAndTriggerNotifications() async{
+    //first we get the .json file that contains the last time that the notification have ran
+    _initFlutterNotificationsPlugin();
+    final notificationStorage = await NotificationTimeoutStorage.create();
+    final userInfo = await AppSharedPreferences.getPersistentUserInfo();
+    final faculties = await AppSharedPreferences.getUserFaculties();
+
+    final Session session =  await NetworkRouter.login(userInfo.item1, userInfo.item2, faculties, false);
+
+    for(Notification Function() value in notificationMap.values){
+        final Notification notification = value();
+        final DateTime lastRan = notificationStorage.getLastTimeNotificationExecuted(notification.uniqueID);
+        if(lastRan.add(notification.timeout).isBefore(DateTime.now())) {
+          await notification.displayNotificationIfPossible(session, _localNotificationsPlugin);
+          notificationStorage.addLastTimeNotificationExecuted(notification.uniqueID, DateTime.now());
+      }
+    }
+  }
+
+  static void initializeNotifications() async{
+    //guarentees that the execution is only done once in the lifetime of the app. 
+    if(_initialized) return;
+    _initFlutterNotificationsPlugin();
+    _buildNotificationWorker();
+    _initialized = true;
+  }
+
+  static void _initFlutterNotificationsPlugin() async{
+    
+    const AndroidInitializationSettings initializationSettingsAndroid = AndroidInitializationSettings('ic_notification');
+
+    //request for notifications immediatly on iOS
+    const DarwinInitializationSettings darwinInitializationSettings = DarwinInitializationSettings(
+      requestAlertPermission: true,
+      requestBadgePermission: true,
+      requestCriticalPermission: true
+    );
+
+    const InitializationSettings initializationSettings = InitializationSettings(
+      android: initializationSettingsAndroid,
+      iOS: darwinInitializationSettings,
+      macOS: darwinInitializationSettings
+    );
+
+    await _localNotificationsPlugin.initialize(initializationSettings);
+
+    //specific to android 13+, 12 or lower permission is requested when the first notification channel opens
+    if(Platform.isAndroid){
+      final AndroidFlutterLocalNotificationsPlugin androidPlugin = _localNotificationsPlugin.resolvePlatformSpecificImplementation()!;
+      try{
+        final bool? permissionGranted = await androidPlugin.requestPermission();
+        if(permissionGranted != true){
+          //FIXME: handle this better
+          throw Exception("Permission not granted for android...");
+        }
```

Comment:
It's begging to be fixed

Reply:
I left this like this, because I was unsure how to handle it. Should something be displayed when the user rejects it, or should it silently return? Keep in mind that this way of asking for notification permissions is quite new and is only available on android 13+. On older versions of android, the permission will be asked when the first notification channel opens with no return to the app, in case the user rejects it.

Reply:
Something like `if !permissionGranted show toast() return` would be ok I think

Reply:
A toast should have a buildcontext to be displayed, which I'm not sure how to get. I think that a silent return to the function is appropriate here.